### ISSUE TITLE
Possibly Stop Using IHttpContextAccessor

### ISSUE THREAD

**Issue by StrangeWill** [RoushTech/RollbarDotNet#33]
As part of this announcement: https://github.com/aspnet/Hosting/issues/793 which caused error #32, the .NET Core team said they removed IHttpContextAccessor because of "non-trivial performance costs".

Is there another option for adding Features to HTTPContext that is better performance wise?

**Comment by morgwai** [comment]
first, as the initial post in #793 says, IHttpContextAccessor has not been removed totally: just not configured in the default case and you can still use it after adding  `services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();`
line to your Configure method (I know that you know, but it's a clarification not to freak out others who may come to this thread from a web search or something ;) )
second, if you read the whole discussion you will find that the statement about "non-trivial performance costs" was premature: the person who wrote it later explained that it was just "Speculation/Question". Unless it is implemented in some "exotic" way it should be just a single additional pointer substitution when a thread finishes handling one request and moves to a new one.

**Comment by StrangeWill** [comment]
Hey @morgwai,

  I really appreciate you doing the leg work on that, I had just tossed this here after skimming the issue to see what I needed to do to be followed on up later, I'll see about adding that line to our service configurator and call it good. I guess being as this will be annoying for anyone running without Identity services that I'll _plan_ on getting a patch out with this in by this weekend (I'll also see about adding test cases to cover these circumstances and all our dependency injection configuration).

  Awesome that the "claim" about the performance hit was more of a question then a claim-- I'd hate to take a significant performance hit for what is just exposing a feature or start adding a bunch of configuration options for this.
